netlify/functions/manage-findings.ts

```diff
@@ -90,44 +90,85 @@ async function getFinding(
   };
 }
 
+async function getTeamsFindings(
+  teamNames: string[],
+  octokit: Octokit,
+  contest: Contest,
+  wardenFindingsForContest: WardenFindingsForContest
+): Promise<void[]> {
+  const promises: Promise<void>[] = [];
+  for (const teamName of teamNames) {
+    promises.push(
+      getSingleTeamFindings(teamName, octokit, contest).then((teamFindings) => {
+        wardenFindingsForContest.teams[teamFindings.teamName] =
```

Comment:
Haven't gotten to the calling code yet but assuming there's a really compelling reason to do this as a side effect then this should be flagged in a comment so it stands out!

Otherwise refactor to more like:

```
async function getTeamsFindings(
  teamNames: string[], octokit: Octokit, contest: Contest
): Promise<WardenFindingsForContest>
```

Reply:
Ah, actually there's two directions here I'm not sure which one you prefer but should probably pick one:

```
let collection;
for (const x of things) {
   const data = await someFetchRelatedTo(x)
   collection[data.thing] = whatever;
}
return collection;
```

Or otherwise:

```
return await Promise.all(
  things.map(gatherThem)
).then((arr) => arr.reduce((collection, data) => {
  collection[data.thing] = whatever
  return collection;
}, {});
```
if you like `reduce` but probably more clearly:

```
return await Promise.all(
  things.map(gatherThem)
).then((arr) => {
  let collection = {};
  arr.forEach((data) => {
    collection[data.thing] = whatever
  });
  return collection;
```

But either way get rid of giving a name to `promises` especially when they don't resolve to anything useful.

Reply:
**UDPATE**: aaaaah forgot the whole point here based on one of the commit messages of this PR, is to make it parallel!


So I misunderstood some code above, but I still maintain that this generically named `promises` is a code smell and we should establish a better pattern for this.

Problem: we're waiting to process an array of input in parallel, using promises
Context: when we [do, or alternately do NOT] know that the array length is small

The solution depends on the expected array size. If you know it's small you can maybe do it ± this way just with a little clearer code. If you know it could be big then you probably do need a better-enapsulated strategy from the start.

Reply:
In probably 99% of cases, the array size is going to be 0 or 1. There are few wardens on multiple teams. Given that context, maybe it's fine to just fetch team findings one at a time, but in parallel with warden findings? Also, moving this function to the api will speed things up because we won't have to wait for a response from the api to get contest data. Ideally that move will happen soon.

Reply:
Nah, my assumption in approving this PR was just that. I think what you have will work for now and I'll file a followup task for myself to propose a longer-term pattern to circle back and clean up the parallel code stuff that's not just here 
but in iirc already other places too.